tests/zenko_tests/node_tests/cloudserver/bucketPolicy/tests/bucketPolicy.js

```diff
@@ -0,0 +1,565 @@
+'use strict'; // eslint-disable-line strict
+
+const assert = require('assert');
+const async = require('async');
+const uuidV4 = require('uuid/v4');
+
+const { scalityS3Client, altScalityS3Client } = require('../../../s3SDK');
+const testUtils = require('../../../utils/testUtils');
+
+const bucket = `bpolicy-bucket-${uuidV4()}`;
+const bParam = { Bucket: bucket };
+const objKey = 'bucket-pol-key';
+const oParam = { Bucket: bucket, Key: objKey };
+const mpuKey = 'bucket-pol-mpu-key';
+const publicId = 'http://acs.amazonaws.com/groups/global/AllUsers';
+const webConfig = {
+    IndexDocument:
+    {
+        Suffix: 'index.html',
+    },
+};
+const corsConfig = {
+    CORSRules: [
+        {
+            AllowedMethods: ['GET'],
+            AllowedOrigins: ['*'],
+        },
+    ],
+};
+const lcConfig = {
+    Rules: [
+        {
+            ID: 'test-id',
+            Status: 'Enabled',
+            Prefix: '',
+            Expiration: {
+                Days: 1,
+            },
+        },
+    ],
+};
+const repConfig = {
+    Role: 'arn:aws:iam::account-id:role/src-resource',
+    Rules: [
+        {
+            Destination: {
+                Bucket: `arn:aws:s3:::${bucket}`,
+                StorageClass: 'STANDARD',
+            },
+            Prefix: 'test-prefix',
+            Status: 'Enabled',
+            ID: 'test-id',
+        },
+    ],
+};
+const vConfig = {
+    Status: 'Enabled',
+};
+
+const basicStatement = {
+    Sid: 'statementid',
+    Effect: 'Allow',
+    Principal: '*',
+    Action: ['s3:GetBucketTagging'],
+    Resource: [
+        `arn:aws:s3:::${bucket}`,
+    ],
+};
+
+const objArn = `arn:aws:s3:::${bucket}/*`;
+const allow = 'allow alt account to perform specified action';
+
+function getPolicyParams(paramsToChange) {
+    const newParam = {};
+    const bucketPolicy = {
+        Version: '2012-10-17',
+        Statement: [basicStatement],
+    };
+    if (paramsToChange) {
+        for (let i = 0; i < paramsToChange.length; i++) {
+            newParam[paramsToChange[i].key] = paramsToChange[i].value;
+            bucketPolicy.Statement[0] = Object.assign(
+                {}, basicStatement, newParam);
+        }
+    }
+    return {
+        Bucket: bucket,
+        Policy: JSON.stringify(bucketPolicy),
+    };
+}
+
+describe('Bucket policies', () => {
+    beforeEach(done => {
+        scalityS3Client.createBucket({
+            Bucket: bucket,
```

Comment:
It might help out down the line debugging tests by generating a new bucket name UUID for each `beforeEach`.